src/plugins/share/server/routes/goto.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CoreSetup, IRouter } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+
+import { handleShortUrlError } from './lib/short_url_error';
+import { shortUrlAssertValid } from './lib/short_url_assert_valid';
+import { ShortUrlLookupService } from './lib/short_url_lookup';
+
+export const createGotoRoute = ({
+  router,
+  shortUrlLookup,
+  http,
+}: {
+  router: IRouter;
+  shortUrlLookup: ShortUrlLookupService;
+  http: CoreSetup['http'];
+}) => {
+  router.get(
+    {
+      path: '/goto/{urlId}',
+      validate: {
+        params: schema.object({ urlId: schema.string() }),
+      },
+    },
+    async function(context, request, response) {
+      try {
+        const url = await shortUrlLookup.getUrl(request.params.urlId, {
+          savedObjects: context.core.savedObjects.client,
+        });
+        shortUrlAssertValid(url);
+
+        const uiSettings = context.core.uiSettings.client;
+        const stateStoreInSessionStorage = await uiSettings.get('state:storeInSessionStorage');
+        if (!stateStoreInSessionStorage) {
+          return response.redirected({
+            headers: {
+              location: http.basePath.prepend(url),
+            },
+          });
+        } else {
+          return response.redirected({
+            headers: {
+              location: http.basePath.prepend('/goto_LP/' + request.params.urlId),
+            },
+          });
+        }
+      } catch (err) {
+        throw handleShortUrlError(err);
```

Comment:
I believe it wouldn't work as expected because
```
Unlike, hapi route handler in the Legacy platform, any exception raised during the handler call will generate 500 Server error response and log error details for further investigation. See below for returning custom error responses.
```
https://github.com/elastic/kibana/blob/master/docs/development/core/server/kibana-plugin-server.httpservicesetup.md

You might want to use https://github.com/elastic/kibana/pull/51157

Reply:
Good catch, thanks. I switched to the wrapper and it works fine. The case it doesn't handle (translating unknown exceptions to 500 errors) seems to be the default behavior of the NP router anyway, so I removed the special error handling completely.